proto/internal/temporal/server/api/persistence/v1/chasm.proto

```diff
@@ -86,8 +99,8 @@ message ChasmComponentRef {
     // reference was created. Can be used to invalidate this reference.
     VersionedTransition component_last_update_versioned_transition = 2;
 
-    // Full path to the component.
-    repeated string path = 3;
+    // Encoded full path to the component.
+    string path = 3;
```

Comment:
Makes the generate task logic a bit easier since we already have the encoded path for all updated nodes and thus avoiding creating the string slices for node names.

When processing the task, we need to do an additional decoding step. However, as the TODO commented added in the PathEncoder interface says, we can do that without any additional allocation by having the Decode method return an iterator.